openrtb_ext/request_wrapper.go

```diff
@@ -0,0 +1,773 @@
+package openrtb_ext
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/mxmCherry/openrtb/v15/openrtb2"
+)
+
+// RequestWrapper wraps the OpenRTB request to provide a storage location for unmarshalled ext fields, so they
+// will not need to be unmarshalled multiple times.
+type RequestWrapper struct {
+	// json json.RawMessage
+	Request *openrtb2.BidRequest
+	// Dirty bool // Probably don't care
+	UserExt    *UserExt
+	DeviceExt  *DeviceExt
+	RequestExt *RequestExt
+	AppExt     *AppExt
+	RegExt     *RegExt
+	SiteExt    *SiteExt
+}
+
+func (rw *RequestWrapper) ExtractUserExt() error {
+	if rw.UserExt != nil {
+		return nil
+	}
+	rw.UserExt = &UserExt{}
+	if rw.Request == nil || rw.Request.User == nil || rw.Request.User.Ext == nil {
+		return rw.UserExt.Unmarshal(json.RawMessage{})
```

Comment:
Since we got rid of having separate Extract() and Unmarshal() methods on the ext object, Unmarshal() now needs to do it all. In this case, Unmarshal() is also tasked with initializing any data structures that need to be initialized, so we call it with nil ext JSONs to ensure any needed initialization is done. I believe currently only 2 objects need this, UserExt and one other that escapes me for the moment.